### ISSUE TITLE
Can’t access ip address interface

### ISSUE THREAD

**Issue by Jack123uk** [rstrouse/ESPSomfy-RTS#184]
Hey, I seem to be having an issue where everything will work great for so long no more then 2 days, then my blinds will become unavailable, if I try and access the ip address of espsomfyrts it will not connect, the only way to resolve the issue is turning the device off and back on again. I would say it’s like it’s lost WiFi connection but when I look in my router settings it’s still online and available. Do you know what the issue may be? Thanks

**Comment by rstrouse** [comment]
Is there any way you can hook it up to a serial connection to see the logs when it fails?  This would mean that you have it plugged into and powered from the serial device.

**Comment by Jack123uk** [comment]
Yes the time zone is correct, I’ve reserved the ip address within my router settings is that what you mean? I’ve plugged it into my computer and launched esphome web to see the logs, 
It says successfully connected to ip address, but I can’t access it still, 
And it seems to get stuck on initializing RX Queue

**Comment by rstrouse** [comment]
The very next thing it does is call out to github to check for the next release.  This is a great clue.  Can your device reach the internet?  I may need to figure out how to fail the request sooner if that is what is failing.

**Comment by Jack123uk** [comment]
Yes I have internet to the device.  After many many reboots I’ve got it all working again, but if it’s following the same trend it will go unavailable again in 24/48 hours. Hopefully not.

**Comment by rstrouse** [comment]
What are you powering with?

**Comment by Jack123uk** [comment]
An Apple usb charger plug.

**Comment by rstrouse** [comment]
It is odd that it is failing like that.  It will be interesting to see the log when it does fail.  I will add some code to timeout if the interweb gets lost.

**Comment by Jack123uk** [comment]
Okie thank you.. yes it’s very odd behaviour. I will see how it goes and see if it does it again..

**Comment by Jack123uk** [comment]
So i went to update the device today and the same as happened after the update, i cannot access the device from the ip address anymore. connecting it to the logs it stops at loading file /main.css.

Radio Pins Configured!
Successfully set up the radio
Enabled receive on Pin #12 Timing: 1
Initializing RX Queue
[HTTPS] GET... code: 200
[HTTPS] GET... code: 200 - 12884
Socket [0] Connected from (my ip address) url: /
Loading file /index.html
Loading file /main.css

**Comment by rstrouse** [comment]
The Loading file /main.css is interesting as the very next thing is should get is a request for the /icons.css file.  I have added a response to the options request just in case the browser asks for it.  The other thing to check is to see if the browser cache is being disrupted.  You will see an error in the developer tools if there is an issue with the browser cache.  There is no error and the there is no follow on request from the browser.

Update the firmware and application from the Main repo and lets see if I see an options request.

**Comment by rstrouse** [comment]
@mvantschip what version were you coming from?

**Comment by rstrouse** [comment]
@mvantschip the issue that @Jack123uk is having is corruption of the SPIFFS partition.  Although, I think your SPIFFS is corrupt I suspect that it is probably related to a quirk in the update library.  

I have had a few folks that have had the application upload process not write the final bytes to the SPIFFS partition because the code for ending the update reboots the MCU before it has had time to flush.  In v2.2.x I have added a delay after the final bytes to allow it to flush.  In all the hundreds of times I updated the SPIFFS partition it happened only once but I suspect it requires the response to process before it forces the reboot.  This allowed me to understand what was going on because the device was connected to the log at the time.

Unfortunately, you will need to flash the SomfyController.onboard.esp32.bin file then restore the backup that was created when you tried to install the SomfyController.littlefs.bin.  v2.2.x and above should no longer have this intermittent issue.
